samples/subsys/usb/hid-cdc/src/main.c

```diff
@@ -0,0 +1,868 @@
+		case 45:
+			return HID_KEY_MINUS;
+		case 46:
+			return HID_KEY_DOT;
+		case 47:
+			return HID_KEY_SLASH;
+		default:
+			return -1;
+		}
+	} else if (ascii < 58) {
+		/* Numbers */
+		if (ascii == 48) {
+			return HID_KEY_0;
+		} else {
+			return ascii - 19;
+		}
+	} else if (ascii < 65) {
+		/* Special characters #2 */
+		switch (ascii) {
+		case 58:
+			return HID_KEY_SEMICOLON;
+		case 59:
+			return HID_KEY_SEMICOLON;
+		case 60:
+			return HID_KEY_COMMA;
+		case 61:
+			return HID_KEY_EQUAL;
+		case 62:
+			return HID_KEY_DOT;
+		case 63:
+			return HID_KEY_SLASH;
+		case 64:
+			return HID_KEY_2;
+		default:
+			return -1;
+		}
+	} else if (ascii < 91) {
+		/* Uppercase characters */
+		return ascii - 61;
+	} else if (ascii < 97) {
+		/* Special characters #3 */
+		switch (ascii) {
+		case 91:
+			return HID_KEY_LEFTBRACE;
+		case 92:
+			return HID_KEY_BACKSLASH;
+		case 93:
+			return HID_KEY_RIGHTBRACE;
+		case 94:
+			return HID_KEY_6;
+		case 95:
+			return HID_KEY_MINUS;
+		case 96:
+			return HID_KEY_GRAVE;
+		}
+	} else if (ascii < 123) {
+		/* Lowercase letters */
+		return ascii - 93;
+	} else if (ascii < 128) {
+		/* Special characters #4 */
+		switch (ascii) {
+		case 123:
+			return HID_KEY_LEFTBRACE;
+		case 124:
+			return HID_KEY_BACKSLASH;
+		case 125:
+			return HID_KEY_RIGHTBRACE;
+		case 126:
+			return HID_KEY_GRAVE;
+		case 127:
+			return HID_KEY_DELETE;
+		default:
+			return -1;
+		}
+	}
+
+	return -1;
+}
+
+static inline bool needs_shift(u8_t ascii)
+{
+	if ((ascii < 33) || (ascii == 39)) {
+		return false;
+	} else if ((ascii >= 33) && (ascii < 44)) {
+		return true;
+	} else if ((ascii >= 44) && (ascii < 58)) {
+		return false;
+	} else if ((ascii == 59) || (ascii == 61)) {
+		return false;
+	} else if ((ascii >= 58) && (ascii < 91)) {
+		return true;
+	} else if ((ascii >= 91) && (ascii < 94)) {
+		return false;
+	} else if ((ascii == 94) || (ascii == 95)) {
+		return true;
+	} else if ((ascii > 95) && (ascii < 123)) {
+		return false;
+	} else if ((ascii > 122) && (ascii < 127)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+/* CDC ACM */
+
+static volatile bool data_transmitted;
+static volatile bool data_arrived;
+
+static void flush_buffer_mouse(void)
+{
+	chr_ptr_mouse = 0;
+	memset(data_buf_mouse, 0, sizeof(data_buf_mouse));
+}
+
+static void flush_buffer_kbd(void)
+{
+	chr_ptr_kbd = 0;
+	memset(data_buf_kbd, 0, sizeof(data_buf_kbd));
+}
+
+static void write_data(struct device *dev, const char *buf, int len)
+{
+	uart_irq_tx_enable(dev);
+
+	while (len) {
+		int written;
+
+		data_transmitted = false;
+		written = uart_fifo_fill(dev, (const u8_t *)buf, len);
+		while (data_transmitted == false) {
+			k_yield();
+		}
+
+		len -= written;
+		buf += written;
+	}
+
+	uart_irq_tx_disable(dev);
+}
+
+static void cdc_mouse_int_handler(struct device *dev)
+{
+	uart_irq_update(dev);
+
+	if (uart_irq_tx_ready(dev)) {
+		data_transmitted = true;
+	}
+
+	if (uart_irq_rx_ready(dev)) {
```

Comment:
Perhaps, it'd be more readable if you `return` early from this function, when the `rx` interrupt is not ready.

Reply:
yes, it could reduce number of indentations.

Reply:
Fixed.